src/build/help/help.xml

```diff
@@ -1085,6 +1085,16 @@
                         <example>~/.ssh/id_ed25519.pub</example>
                     </config-key>
 
+                    <config-key id="repo-sftp-req-known-hosts-match" name="Require a known_hosts match for the SFTP host">
```

Comment:
Instead of a yes/no flag and a hard-coded .ssh/known_hosts file, we should be following closer to what OpenSSH does and instead have an option where the user can tell us of the known_hosts files to use, and then also have a default which will automatically be used if nothing has been specified.

eg: repo-sftp-known-hosts-file="/etc/ssh/ssh_known_hosts /etc/ssh/ssh_known_hosts2 ~/.ssh/known_hosts ~/.ssh/known_hosts2"

would be the default, while explicitly specifying the file is supported in the same way and specifying it but leaving it blank would disable the feature.

Also- we don't need both this and the explicit host fingerprint.  The explicit fingerprint should be used if it's set and perhaps we should throw a warning if a known hosts file and an explicit fingerprint are configured because it could lead to confusing results.  In other words- if *both* are explicitly set, then throw a warning or error saying that it doesn't make sense to have them both set.  If *only* the fingerprint is set, then use only that and treat the known hosts file list as empty.  If *only* the known host file list is set, then use that and we shouldn't require the host fingerprint to be set (I don't think we do, but just trying to be clear).

Reply:
If *neither* are set, then we should use the default known_hosts files that I included in the above comment, matching how OpenSSH works today.

Reply:
Implemented in commit ccacee0b3640f9fa173b70190540927695b911bd. I believe that the only discrepancy is that repo-sftp-known-hosts-files can't be configured as empty to disable, so implemented as an explicit disable repo-sftp-known-hosts-files=disable